### PR TITLE
fix(1.14/publish): fix publish release

### PR DESCRIPTION
**Pull request by Han-Ya-Jun**
### Description
fix(1.14/publish): fix publish release

Fixes # (issue)

### Checklist

- [x] 填写 PR 描述及相关 issue (write PR description and related issue)
- [x] 代码风格检查通过 (code style check passed)
- [x] PR 中包含单元测试 (include unit test)
- [x] 单元测试通过 (unit test passed)
- [x] 本地开发联调环境验证通过 (local development environment verification passed)

### CODE REVIEW

File: src/dashboard/apigateway/apigateway/controller/tasks/release.py [hunk 1]
```diff
@@ -146,7 +147,32 @@ def release_gateway_by_registry(
         micro_gateway_id,
         micro_gateway_release_history_id,
     )
```
Comment (wklken) [inline src/dashboard/apigateway/apigateway/controller/tasks/release.py:149]: 1. 去掉release_id
2. publish_id变成必填
3. 不存在publish_id / release_history， 直接抛异常（不会出现）

File: src/dashboard/apigateway/apigateway/controller/tasks/release.py [hunk 2]
```diff
@@ -146,7 +147,32 @@ def release_gateway_by_registry(
...
+        return None
+
+    # 改成了延迟更新发布关联数据，这里的release数据需要构造才行
+    release = Release.objects.save_release(
+        gateway=release_history.gateway,
+        stage=release_history.stage,
+        resource_version=release_history.resource_version,
+        comment=release_history.comment,
+        username=release_history.created_by,
+    )
```
Comment (wklken) [inline src/dashboard/apigateway/apigateway/controller/tasks/release.py:175]: 这里不需要构造一个fake release， 成功后再保存（避免提前建立关系）

**Review by wklken — APPROVED** [review]

File: src/dashboard/apigateway/apigateway/controller/tasks/release.py (hunk 1 above)
Comment (Han-Ya-Jun) [reply src/dashboard/apigateway/apigateway/controller/tasks/release.py:145]: done

File: src/dashboard/apigateway/apigateway/controller/tasks/release.py (hunk 2 above)
Comment (Han-Ya-Jun) [reply src/dashboard/apigateway/apigateway/controller/tasks/release.py:163]: done

**Pull request merged**